src/lib/util/AntiUserGateway.ts

```diff
@@ -0,0 +1,225 @@
+		return 'Map';
+	}
+
+	public increase(): Promise<SettingsUpdateResults> {
+		return Promise.resolve([]);
+	}
+
+	public decrease(): Promise<SettingsUpdateResults> {
+		return Promise.resolve([]);
+	}
+
+	public clone(): Settings {
+		return Object.create(this);
+	}
+
+	public sync(): Promise<this> {
+		return Promise.resolve(this);
+	}
+
+	public destroy(): Promise<this> {
+		return Promise.resolve(this);
+	}
+
+	public get client() {
+		return this.gateway.client;
+	}
+
+	public get(): unknown {
+		return undefined;
+	}
+
+	public pluck(): unknown[] {
+		return [];
+	}
+
+	public resolve(): Promise<unknown[]> {
+		return Promise.resolve([]);
+	}
+
+	public reset(): Promise<SettingsUpdateResults> {
+		return Promise.resolve([]);
+	}
+
+	public update(): Promise<SettingsUpdateResults> {
+		return Promise.resolve([]);
+	}
+
+	public toJSON(): SettingsFolderJson {
+		return {};
+	}
+
+	public clear(): void {
+		// noop
+	}
+
+	public delete(): boolean {
+		return false;
+	}
+
+	public forEach(): void {
+		// noop
+	}
+
+	public has(): boolean {
+		return false;
+	}
+
+	public set(): this {
+		return this;
+	}
+
+	public [Symbol.iterator](): IterableIterator<[string, unknown]> {
+		throw new Error('Method not implemented.');
+	}
+
+	public entries(): IterableIterator<[string, unknown]> {
+		throw new Error('Method not implemented.');
+	}
+
+	public keys(): IterableIterator<string> {
+		throw new Error('Method not implemented.');
+	}
+
+	public values(): IterableIterator<unknown> {
+		throw new Error('Method not implemented.');
+	}
+
+	protected _patch(): void {
+		// noop
+	}
+
+	protected _init(): void {
+		// noop
+	}
+
+	protected _save(): Promise<void> {
+		return Promise.resolve();
+	}
+
+	protected _resolveFolder(): void {
+		// noop
+	}
+
+	protected _resolveEntry(): void {
+		// noop
+	}
+
+	protected _resetSettingsFolder(): void {
+		// noop
+	}
+
+	protected _resetSettingsEntry(): void {
+		// noop
+	}
+
+	protected _processUpdate(): void {
+		// noop
+	}
+
+	protected _updateSettingsEntry(): void {
+		// noop
+	}
+
+	protected _updateSettingsEntryNotIndexed(): void {
+		// noop
+	}
+
+	protected _updateSettingsEntryAtIndex(): void {
+		// noop
+	}
+
+	protected _resolveValues(): void {
+		// noop
+	}
+
+	protected _resolveNextValue(): void {
+		// noop
+	}
+
+	protected _updateSchemaEntryValue(): void {
+		// noop
+	}
+
+}
+
+// Step 3: Create UserGateway, which is an empty Gateway-like class, implements:
+//         Gateway.
+export class UserGateway extends GatewayStorage implements Gateway {
+
+	public cache = new Map();
+	public requestHandler = new RequestHandler<string, IdKeyed<string>>(
+		id => Promise.resolve({ id }),
+		ids => Promise.resolve(ids.map(id => ({ id })))
+	);
+
+	public acquire(): Settings {
+		return this.create();
+	}
+
+	public get(): Settings | null {
+		return null;
+	}
+
+	public create(): Settings {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
```

Comment:
Add a short description as to why there is a ts-ignore then eslint won't complain either.